classes/Task/AbstractTask.php

```diff
@@ -152,7 +159,7 @@ public function setErrorFlag(): void
         if (static::TASK_TYPE) {
             $this->container->getAnalytics()->track(
                 ucfirst(static::TASK_TYPE) . ' Failed',
-                static::TASK_TYPE === 'upgrade' ? Analytics::WITH_UPGRADE_PROPERTIES : Analytics::WITH_ROLLBACK_PROPERTIES
+                static::TASK_TYPE === TaskType::TASK_TYPE_UPGRADE ? Analytics::WITH_UPGRADE_PROPERTIES : Analytics::WITH_ROLLBACK_PROPERTIES
```

Comment:
We need to improve this test for the 3 possible values. We will split options for backup and update from now.